### ISSUE TITLE
Add an API to create an environment tag

### ISSUE THREAD

**Issue by gashcrumb** [syndesisio/syndesis#5263]
## This is a...

<pre><code>
[x] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Documentation issue or request
</code></pre>

## Description
There's APIs to get, rename and delete environments, unfortunately there isn't an API to create one.

For the moment in [this PR](https://github.com/syndesisio/syndesis-react/pull/161) I'm just sending the request to the rename API so the UI doesn't blow up when you try and add a new environment.

@dhirajsb if you get a chance this'd be nice...

**Comment by mkralik3** [comment]
I have changed the priority to p0 because the UI is now in the master and CI/CD feature doesn't work without this feature request. The user is not able to create any tag in the UI.

**Comment by tplevko** [comment]
@heiko-braun @dhirajsb, this was brought to my attention. What is the plan for this feature? The CI/CD functionality from 7.3. is not working ATM.

**Comment by heiko-braun** [comment]
@mkralik3 @gashcrumb Is this something missing in react, or how does it differ from 7.3?

**Comment by heiko-braun** [comment]
@dhirajsb How did we create environments in 7.3?

**Comment by mkralik3** [comment]
@heiko-braun If I understand it correctly, the process of how to manage **tags** was changed a little. 
Now, the tags are managed in the separate UI page (_Manage CI/CD_).
![image](https://user-images.githubusercontent.com/16251792/59331658-c36b2a80-8cf4-11e9-9be5-f1c9ae5c7ab3.png)
It is needed to add the ability for creating a tag in this page. And for it, we need a new endpoint to create a tag which is not associated with any integration. 
In the dialog on the particular integration, a user can only check or uncheck existed tags now.
![image](https://user-images.githubusercontent.com/16251792/59331684-d1b94680-8cf4-11e9-990f-e9496904627d.png)

This is different from the **7.3** where the user has to create **tag only in** the dialog on the particular integration. 
![image](https://user-images.githubusercontent.com/16251792/52949215-f3747380-337b-11e9-846e-e14abffc2438.png)
So every time when the tag was created, it was associated to the at least one integration. Because of that, the endpoint for creating a tag which is not associate to any integration was not needed. 
And if it had existed in **7.3**, it would not have worked because when the tag without any associated integration would be created, it would be automatically deleted. For this behavior is another issue  -> https://github.com/syndesisio/syndesis/issues/5266
 
In short, in the **7.3** the new tag is created for the particular integration.
`PATCH ​/public​/integrations​/{id}​/tags -d '["tag6","tag7"]`

**Comment by gashcrumb** [comment]
I've already stated that the new UI needs 3 adjustments (hence the 3 separate issues) to the API to work properly because they're simply not there or don't work in the way that's expected based on the new design.

Basically if we don't adjust the API then we may as well hide the manage CI/CD button and links because ultimately the feature will ship broken.  Or just ship the angular console this release, as that's the UI the API is currently designed to support.

**Comment by dhirajsb** [comment]
Did we decide to treat environments as standalone entities for 7.4? It requires model redesign that I am working on but didn't finish before the feature freeze. 
The model redesign also depends on other features that need to be included like Fuse variables (a separate feature that was deferred) and would require UXD. 

To make this work, I could add a stub create API for environment that simply caches environment names in memory, to allow users to create the names in the Manage CI/CD page, and then associate them with integrations.

**Comment by gashcrumb** [comment]
So this issue and [this one](https://github.com/syndesisio/syndesis/issues/5266) are the main gaps with the UI that's based on the updated design.  When you say "model redesign" I want to hide under a rock.

**Comment by dhirajsb** [comment]
Yeah, as @mkralik3 mentioned earlier, the current model design 'hangs' environments names off integrations as just string labels with a few properties. If we want to treat environments as first class entities, it requires a slight model refactor around environments. But it will affect a lot of backend code for the Public API. 

The in-memory cached names solution can satisfy both this issue and #5266. The only inconvenience is that unassigned/unused tags will only live in server memory and will be dropped if the server is restarted. Considering that it's a relatively very low percentage event, the user will see consistent behavior for all practical purposes.

**Comment by gashcrumb** [comment]
That seems reasonable to me for this version. cool!

**Comment by heiko-braun** [comment]
@dhirajsb I'd suggest we move forward with your proposal in 7.4 and create a followup ticket for 7.5

**Comment by mkralik3** [comment]
@dhirajsb Is this already done? The syndesis-bot marks this issue as Done, however, I don't see the commit in the master.

**Comment by dhirajsb** [comment]
@mkralik3 I see the commit on master at https://github.com/syndesisio/syndesis/commit/ec9cd881de005081c8ac57d47d47856dd58513a0

**Comment by dhirajsb** [comment]
@mkralik3 I found a slight error in the UI and the new create environment API. I'm going to push a PR to fix that tonight.

**Comment by dhirajsb** [comment]
Created #5736 for backend and https://github.com/syndesisio/syndesis-react/pull/473 to fix create API call.

**Comment by mkralik3** [comment]
Verified. The endpoint works. However, for the UI part, this issue needs to be resolved:  https://github.com/syndesisio/syndesis/issues/5828
